src/Sarif.Multitool/Rules/SARIF1002.UrisMustBeValid.cs

```diff
@@ -62,19 +62,6 @@ protected override void Analyze(ReportingDescriptor reportingDescriptor, string
             AnalyzeUri(reportingDescriptor.HelpUri, messageDescriptorPointer.AtProperty(SarifPropertyName.HelpUri));
         }
 
-        protected override void Analyze(Run run, string runPointer)
```

Comment:
>protected override void Analyze [](start = 8, length = 31)

see `SarifValidationSkimmerBase`. For 2003 to look at all possible artifactLocations, i had to add a Visit to OriginalUriBaseIds (it was missing!).

So, in 1002, we dont need this function anymore, there is already an analyzer method for ArtifactLocations in this rule, which will now cover this one as well.

Reply:
But the new analyzer does not ensure that the URI is valid. Restore this code.

---
In reply to: [449326625](https://github.com/microsoft/sarif-sdk/pull/1966#discussion_r449326625) [](ancestors = 449326625)

Reply:
Larry, this does not change functionality - this rule is still looking at OriginalUriBaseIds since they are now being visited via `Analyze(ArtifactLocation fileLocation, string fileLocationPointer)` method on line 41. we just dont need this additional function anymore.

---
In reply to: [449328340](https://github.com/microsoft/sarif-sdk/pull/1966#discussion_r449328340) [](ancestors = 449328340,449326625)

Reply:
Understood! Let's keep this change and revert the rest of this PR. I need to re-author the rule descriptor for 1004, and the result message for UriBaseIdRequiresRelativeUri.

---
In reply to: [449329017](https://github.com/microsoft/sarif-sdk/pull/1966#discussion_r449329017) [](ancestors = 449329017,449328340,449326625)